packages/did-ethr-resolver/src/implementations/ewSigner.ts

```diff
@@ -104,8 +103,19 @@ export class EwSigner extends Signer {
   * @param eip1993Provider an EIP1193 provider (https://docs.ethers.io/v5/api/providers/other/#Web3Provider)
   * @param publicKey the publicKey of the signer associated with the provider
   */
-  public static fromEIP1193(eip1993Provider: EIP1193ProviderType, publicKey: string): EwSigner {
+  // eslint-disable-next-line max-len
+  public static async fromEIP1193(eip1993Provider: EIP1193ProviderType, publicKey: string): Promise<EwSigner> {
     const provider = new providers.Web3Provider(eip1993Provider);
-    return new EwSigner(provider.getSigner(), publicKey);
+    const signer = provider.getSigner();
+    try {
+      // Call getAddress() to check that signer has address
+      // Signer could be returned by not actually have an address
+      // Best to check now and "fail fast"
+      await signer.getAddress();
```

Comment:
Signer can have address event not being connected, for example Wallet. Better check provider set or chain id

Reply:
```suggestion
      await signer.getChainId();
```

Reply:
Also it is better to put this check in `EwSigner` constructor to not duplicate it in all factories

Reply:
- In this case, I think we know that the `signer` has a `provider` because the `signer` is coming from `provider.getSigner()`
- I don't think it can be put in the constructor as it is an `async` call?

Reply:
Right. Initially I misunderstood purpose of this. Its ok then. Though I think it is not responsibility of the did-reg to verify correctness of the signer implementation